### PR TITLE
Feature/668 add reference to bpi account in a workflow

### PR DESCRIPTION
**Pull request by ognjenkurtic**
# Description

Introduces Bpi Account creation during creation of a Workflow, with a reference Workflow -> BpiAccount.

## Related Issue

#668 

## Motivation and Context

It allows for workflows to store state inside a BpiAccount which is co owned by all workgroup participants

## How Has This Been Tested

Partially by unit test. Full unit tests will come once we introduce https://www.prisma.io/docs/guides/testing/unit-testing

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Request to be added as a Code Owner/Maintainer

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I commit to abide by the Responsibilities of Code Owners/Maintainers.

### CODE REVIEW

File: examples/bri-3/src/bri/workgroup/workflows/models/workflow.ts [hunk 1]
```diff
@@ -14,16 +15,24 @@ export class Workflow {
...
+
+  @AutoMap(() => BpiAccount)
+  bpiAccount: BpiAccount;
+
   constructor(
     id: string,
     name: string,
     worksteps: Workstep[],
     workgroupId: string,
+    bpiAccount: BpiAccount,
```
Comment (Therecanbeonlyone1969) [inline examples/bri-3/src/bri/workgroup/workflows/models/workflow.ts:29]: @ognjenkurtic should this not also contain the `bpiAccountId`?

File: examples/bri-3/src/bri/workgroup/workflows/models/workflow.ts [hunk 2]
```diff
@@ -14,16 +15,24 @@ export class Workflow {
...
   ) {
     this.id = id;
     this.name = name;
     this.worksteps = worksteps;
     this.workgroupId = workgroupId;
+    this.bpiAccount = bpiAccount;
```
Comment (Therecanbeonlyone1969) [inline examples/bri-3/src/bri/workgroup/workflows/models/workflow.ts:35]: same here

File: examples/bri-3/src/bri/workgroup/workgroups/agents/workgroupStorage.agent.ts [hunk 3]
```diff
@@ -19,7 +19,7 @@ export class WorkgroupStorageAgent extends PrismaService {
         administrators: true,
         participants: true,
         workflows: {
-          include: { worksteps: true },
+          include: { worksteps: true, bpiAccount: true },
```
Comment (Therecanbeonlyone1969) [inline examples/bri-3/src/bri/workgroup/workgroups/agents/workgroupStorage.agent.ts:22]: `bpiAccountId`?

Comment (Therecanbeonlyone1969) [reply examples/bri-3/src/bri/workgroup/workgroups/agents/workgroupStorage.agent.ts:22]: slightly confused by the difference of bpiAccount which should be an object and bpiAccountId which should be a string

**Review by Kasshern — APPROVED** [review]

File: examples/bri-3/src/bri/workgroup/workflows/models/workflow.ts (hunk 1 above)
Comment (ognjenkurtic) [reply examples/bri-3/src/bri/workgroup/workflows/models/workflow.ts:33]: Good point. Usage of both bpiAccountId and bpiAccount (and this pattern is used across the codebase whenever there is a reference to an object) is not necessary. It is a convention we picked up from how prisma.schema objects are defined. I added a commit that updates bpiAccountId just in the ctor just to be consistent, will have to change this as part of a separate issue that i ll open now.

File: examples/bri-3/src/bri/workgroup/workgroups/agents/workgroupStorage.agent.ts (hunk 3 above)
Comment (ognjenkurtic) [reply examples/bri-3/src/bri/workgroup/workgroups/agents/workgroupStorage.agent.ts:22]: Just a convention picked up from prisma.schema. I do not think we need both of these in the domain object (object reference should be enough) but i cant change that right now across the whole codebase. Let's just be consistent in how we do it and i ll open a separate issue to tackle it.

File: examples/bri-3/src/bri/workgroup/workflows/models/workflow.ts (hunk 1 above)
Comment (ognjenkurtic) [reply examples/bri-3/src/bri/workgroup/workflows/models/workflow.ts:33]: https://github.com/eea-oasis/baseline/issues/719

**Review by Therecanbeonlyone1969 — APPROVED** [review]
LGTM

**Review by Therecanbeonlyone1969 — APPROVED** [review]
LGTM

**Pull request merged**